proj_4.c

```diff
@@ -114,9 +114,9 @@ void writer(int pid)
 	printf("\nThis is the %d th writer writing value i = %d", -pid, ++resource_val);
 	writerExit();
 	yield();
-	readerEntry();
+	writerEntry();
```

Comment:
you can try readerEntry readerExit also they make more sense I was just trying different things